quartz-plugins/src/main/java/org/quartz/plugins/interrupt/JobInterruptMonitorPlugin.java

```diff
@@ -0,0 +1,167 @@
+        return "JobInterruptMonitorListner";
+    }
+
+    public void triggerFired(Trigger trigger, JobExecutionContext context) {
+        // Call the scheduleJobInterruptMonitor and capture the ScheduledFuture in context
+        try {
+            // Schedule Monitor only if the job wants AutoInterruptable functionality
+            if (context.getJobDetail().getJobDataMap().getBoolean(PluginConstants.AUTO_INTERRUPTIBLE)) {
+                JobInterruptMonitorPlugin monitorPlugin = (JobInterruptMonitorPlugin) context.getScheduler()
+                        .getContext().get(JOB_INTERRUPT_MONITER_KEY);
+                // Get the MaxRuntime from Job Data if NOT available use DEFAULT_MAX_RUNTIME from Plugin Configuration
+                long jobDataDelay  = DEFAULT_MAX_RUNTIME;
+
+                if (context.getJobDetail().getJobDataMap().get(PluginConstants.MAX_RUN_TIME) != null){
+                     jobDataDelay = context.getJobDetail().getJobDataMap().getLong(PluginConstants.MAX_RUN_TIME);
+                }
+                future = monitorPlugin.scheduleJobInterruptMonitor(context.getJobDetail().getKey(), jobDataDelay);
+                getLog().debug("Job's Interrupt Monitor has been scheduled to interrupt with the delay :"
+                        + DEFAULT_MAX_RUNTIME);
+            }
+        } catch (SchedulerException e) {
+            getLog().info("Error scheduling interrupt monitor " + e.getMessage(), e);
+        }
+    }
+
+    public void triggerComplete(Trigger trigger, JobExecutionContext context,
+            CompletedExecutionInstruction triggerInstructionCode) {
+        // cancel the Future if job is complete
+        if (future != null) {
+            future.cancel(true);
+        }
+    }
+
+    @Override
+    public void initialize(String name, Scheduler scheduler, ClassLoadHelper helper) throws SchedulerException {
+
+        getLog().info("Registering Job Interrupt Monitor Plugin");
+        this.executor = Executors.newScheduledThreadPool(1);
+        scheduler.getContext().put(JOB_INTERRUPT_MONITER_KEY, this);
+        this.scheduler = scheduler;
+        // Set the trigger Listener as this class to the ListenerManager here
+        this.scheduler.getListenerManager().addTriggerListener(this);
+
+    }
+
+    static class InterruptMonitor implements Runnable {
+
+        private final JobKey jobKey;
+        private final Scheduler scheduler;
+
+        private final Logger log = LoggerFactory.getLogger(getClass());
+
+        InterruptMonitor(JobKey jobKey, Scheduler scheduler) {
+            this.jobKey = jobKey;
+            this.scheduler = scheduler;
+        }
+
+        protected Logger getLog() {
+            return log;
+        }
+
+        @Override
+        public void run() {
+            try {
+
+                // Interrupt the job here - using Scheduler API that gets propagated to Job's interrupt
+                getLog().info("Interrupting Job " + jobKey.getName() + ":" + jobKey.getGroup());
```

Comment:
I wonder if it would be nice to add some words about why it is being interrupted (configured max run time exceeded).